File: programmers/BruteForce/카펫.py
# 프로그래머스 카펫
# 21.02.15

import sys
from functools import cmp_to_key
from collections import deque
from itertools import combinations
from itertools import permutations
sys.setrecursionlimit(10**6)

def solution(brown, yellow):
    answer = []
    b = (brown + 4) // 2
    c = brown + yellow

    answer.append((b + ((b ** 2) - 4 * c) ** 0.5) // 2)
    answer.append((b - ((b ** 2) - 4 * c) ** 0.5) // 2)

    return answer

File: programmers/BruteForce/카펫.swift
// 프로그래머스 카펫
// 21.02.18

import Foundation

func solution(_ brown:Int, _ yellow:Int) -> [Int] {
  let xy = brown + yellow
  var x = brown + yellow
  var y = xy / x
  
  while true {
    if xy % x == 0 && (x - 2) * (y - 2) == yellow { break }
    x -= 1
    y = xy / x
  }
  return [x, y]
}

print(solution(24, 24))
